### PR TITLE
Milestone 3 : Create interactive list

### PR DESCRIPTION
**Pull request by zdnahom**
## Milestone 3 : Create interactive list
In this milestone I : 
- Added new Javascript file 
- Added update status function to the newly added file and exported it
- Updated items object's value for completed key upon user actions(when checkbox is clicked).
- Implemented a function for the "Clear all completed" button to clear out completed todos.
- Stored filtered array to localstorage when checkbox is clicked and when clear button is clicked
- Implemented UI interactions.

### CODE REVIEW

File: src/modules/updateStatus.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  setTodos(todos);
+};
+export const clearTodoList = () => {
+  const listToNotClear = getTodos().filter((todo) => todo.completed === false);
+  const listToClear = getTodos().filter((todo) => todo.completed === true);
+  listToClear.forEach((todo) => {
+    UI.remove(todo.id);
+  });
+  setTodos(listToNotClear);
+};
```
Comment (Gambit142) [inline src/modules/updateStatus.js]: - [ ] After clearing all completed tasks, kindly reset the `index` property of each task to the length of the current array. This way, your tasks are uniform and there won't be two tasks with the same `index` property 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @zdnahom 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Pull request summary is descriptive ✔️ 
- There is a function to update the status of todos ✔️ 
- There is a function to clear all completed tasks ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/modules/updateStatus.js (hunk 1 above)
Comment (zdnahom) [reply src/modules/updateStatus.js]: @Gambit142 Thanks for your insight 👍 . I will do that 🙏

**Review by jerryowusu — APPROVED** [review]
Hi @zdnahom  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

CookedDelectableDolphin-size_restricted


## Highlights
No linter errors :heavy_check_mark: 
Nice commit messages :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**